spec/custom_index_spec.rb

```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
+
+# typed: false
+
+describe "Custom index" do
```

Comment:
woohoo! nice coverage.

can we also test the querying building? for example
```ruby
rel1 = klass.where(indexed_value: ...)
rel1.to_a == rel1.with_index(:first).to_a
rel2 = rel1.where(time_value: ...)
rel2.with_index(:first).to_a
```
and what about following cases:
```
rel1.with_index(:first).with_index(:second)
rel1.with_index(nil) # reset to not use index?
```

Reply:
another test case worth covering:
```
klass.where(indexed_value: ..., value: ...).with_index(:first)
```

Reply:
> klass.where(indexed_value: ..., value: ...).with_index(:first)

I added this particular case, but it still falls in the 'incorrect order of conditions in the query' bucket

Reply:
> ```ruby
> rel1 = klass.where(indexed_value: ...)
> rel1.to_a == rel1.with_index(:first).to_a
> rel2 = rel1.where(time_value: ...)
> rel2.with_index(:first).to_a
> ```
Added this case

Reply:
> ```
> rel1.with_index(:first).with_index(:second)
>```
This would just change custom index in the relation from first to second
>```
> rel1.with_index(nil) # reset to not use index?
> ```
Do you think we need to support this case?

Reply:
> > ```
> > rel1.with_index(nil) # reset to not use index?
> > ```
> 
> 
> Do you think we need to support this case?

if that's easy to support, sure!

Reply:
sounds good